src/modules/deal/dto/deals-request.dto.ts

```diff
@@ -0,0 +1,73 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsDateString, IsNotEmpty, IsNumber, IsOptional, IsString, MaxLength } from 'class-validator';
+
+export class CreateDealDto {
+  @ApiProperty({
+    example: '0020900035', // Comment on PR if we need a description when it's self evident
+  })
+  @IsString()
+  @IsNotEmpty()
+  @MaxLength(10)
+  dealIdentifier: string;
+
+  @ApiProperty({
+    description: 'Deal portfolio is E1',
+    example: 'E1',
+  })
+  @IsString()
+  @IsOptional()
+  @MaxLength(2)
+  portfolioIdentifier = 'E1';
+
+  @ApiProperty({
+    description: 'Deal currency code is USD, CAD',
+    example: 'GBP',
+  })
+  @IsString()
+  @IsNotEmpty()
+  @MaxLength(3)
```

Comment:
Please add `@Min(x)`

Reply:
For currency we always use 3 characters.
Most likely this validator should work @IsISO4217CurrencyCode() from 'class-validator';

Reply:
Thanks @avaitonis, I'll update the validators. Again -- is there a source for information about length restrictions on these fields _other_ than the API specification in AnyPoint? Because I could not see anything other than a maximum value, as in the screenshot above.

Reply:
1. This is ACBS POST Deal documentation -> https://uat-api-uk.acbsasp.com/Help/Api/POST-Portfolio-portfolioIdentifier-Deal
2. ACBS Currency object -> https://uat-api-uk.acbsasp.com/Help/ResourceModel?modelName=Currency
3. Currency objects are saved in -> Currency in the Currency Definition Table
4. CurrencyCode field -> "String length: inclusive between 0 and 3"

If I do GET ACBS /Lists/Currency  , I can see that all currency codes are 3 characters length.

Let me know if you want to get all currencies, but you can find GET ACBS /Lists/Currency in Postman collection 

**Other fields**
Other fields like dealInitialStatus might point to Common Table Maintenance Tables, like T1540. dealInitialStatusCode 
has validation rule: String length: inclusive between 0 and 1

Reply:
Thanks!